pkg/build/api/v1/types.go

```diff
@@ -176,8 +176,46 @@ type BuildStatus struct {
 
 	// output describes the Docker image the build has produced.
 	Output BuildStatusOutput `json:"output,omitempty" protobuf:"bytes,10,opt,name=output"`
+
+	// steps contains details about each build step.
+	Steps Steps `json:"steps,omitempty" protobuf:"bytes,11,opt,name=steps"`
```

Comment:
here's the jenkins metadata we record into the build object.  it has "stages" and "stageflownodes" within the stages, which appear to be the steps.

for both stages and stageflownodes/steps it provides:
startTimeMillis
durationMillis
pauseTimeMillis

```
openshift.io/jenkins-status-json: '{"_links":{"self":{"href":"https://jenkins-c1.router.default.svc.cluster.local/job/c1-sample-pipeline/1/wfapi/describe"}},"id":"1","name":"#1","status":"SUCCESS","startTimeMillis":1489611336514,"endTimeMillis":0,"durationMillis":0,"queueDurationMillis":97492,"pauseDurationMillis":0,"stages":[{"_links":{"self":{"href":"https://jenkins-c1.router.default.svc.cluster.local/job/c1-sample-pipeline/1/execution/node/6/wfapi/describe"}},"id":"6","name":"build","execNode":"","status":"SUCCESS","startTimeMillis":1489611434006,"durationMillis":42214,"pauseDurationMillis":0,"stageFlowNodes":[{"_links":{"self":{"href":"https://jenkins-c1.router.default.svc.cluster.local/job/c1-sample-pipeline/1/execution/node/7/wfapi/describe"}},"id":"7","name":"Trigger
        OpenShift Build","execNode":"","status":"SUCCESS","startTimeMillis":1489611434138,"durationMillis":42074,"pauseDurationMillis":0,"parentNodes":["6"]}]},{"_links":{"self":{"href":"https://jenkins-c1.router.default.svc.cluster.local/job/c1-sample-pipeline/1/execution/node/11/wfapi/describe"}},"id":"11","name":"deploy","execNode":"","status":"SUCCESS","startTimeMillis":1489611476231,"durationMillis":20327,"pauseDurationMillis":0,"stageFlowNodes":[{"_links":{"self":{"href":"https://jenkins-c1.router.default.svc.cluster.local/job/c1-sample-pipeline/1/execution/node/12/wfapi/describe"}},"id":"12","name":"Trigger
        OpenShift Deployment","execNode":"","status":"SUCCESS","startTimeMillis":1489611476253,"durationMillis":20299,"pauseDurationMillis":0,"parentNodes":["11"]}]}]}'
```

So if we want to mirror the jenkins api, we should add the following struct to our build objects:
```
Stage[] {
  name 
  startTime
  duration
  Steps[] {
    name
    startTime
    duration
}
```

@smarterclayton @jwforres tentative agreement?